### ISSUE TITLE
Move function dispatch out of system.entity.base

### ISSUE THREAD

**Issue by ne0ndrag0n** [ne0ndrag0n/Concordia#78]
The existing method of function dispatch (cycle through entities and check their `_sys._sched` table for registered functions on the current tick) is subpar design. Namely, the only thing that can take advantage of function dispatch in Concordia is a `system.entity.base`, rendering system-level UI elements unable to simply slap in a function call for dispatch.

`BlueBear::Engine` should provide a Javascript-like function dispatch loop that allows *any* type of object to register a function for dispatch. This should be either a message queue (where delayed messages are not added to the queue until a tick happens), or a different method where there is only one `_sys._sched`-type table that is shared globally and used by global functions.

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop

**Comment by ne0ndrag0n** [comment]
While implementing the serializer (to overhaul how function dispatch in general is done), I started encountering spurious segmentation faults. Valgrind just vomits endless amounts of data that is not useful. I hate C++.

I opened https://github.com/ne0ndrag0n/Concordia/issues/81 because I believe the issue is thread synchronization. The small bits of info I did find in valgrind's dump looked like it was occurring in the utterly terrible threading that was done. If a segfault happens, ignore it for now. It's occurring when the game closes and frees the thread list.
